File: src/main/java/com/skyspacefivefiftyfive/service/TerminalService.java
package com.skyspacefivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefivefiftyfive.domain.Terminal;
import com.skyspacefivefiftyfive.dto.TerminalDTO;
import com.skyspacefivefiftyfive.dto.TerminalSearchDTO;
import com.skyspacefivefiftyfive.dto.TerminalPageDTO;
import com.skyspacefivefiftyfive.dto.TerminalConvertCriteriaDTO;
import com.skyspacefivefiftyfive.service.GenericService;
import com.skyspacefivefiftyfive.dto.common.RequestDTO;
import com.skyspacefivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface TerminalService extends GenericService<Terminal, Integer> {

	List<Terminal> findAll();

	ResultDTO addTerminal(TerminalDTO terminalDTO, RequestDTO requestDTO);

	ResultDTO updateTerminal(TerminalDTO terminalDTO, RequestDTO requestDTO);

    Page<Terminal> getAllTerminals(Pageable pageable);

    Page<Terminal> getAllTerminals(Specification<Terminal> spec, Pageable pageable);

	ResponseEntity<TerminalPageDTO> getTerminals(TerminalSearchDTO terminalSearchDTO);
	
	List<TerminalDTO> convertTerminalsToTerminalDTOs(List<Terminal> terminals, TerminalConvertCriteriaDTO convertCriteria);

	TerminalDTO getTerminalDTOById(Integer terminalId);







}

File: skyspacefivefiftyfive_ui/src/services/TerminalService.js
import http from "../http-common"; 

class TerminalService {
  getAllTerminals(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/terminal/terminals`, searchDTO);
  }

  get(terminalId) {
    return this.getRequest(`/terminal/${terminalId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/terminal?field=${matchData}`, null);
  }

  addTerminal(data) {
    return http.post("/terminal/addTerminal", data);
  }

  update(data) {
  	return http.post("/terminal/updateTerminal", data);
  }
  
  uploadImage(data,terminalId) {
  	return http.postForm("/terminal/uploadImage/"+terminalId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new TerminalService();
